### PR TITLE
feat: Adding Pulsar sensor

### PR DESCRIPTION
**Pull request by arnarg**
Checklist:

* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-events/blob/master/USERS.md).

Adds a Pulsar sensor trigger so you can send messages to a Pulsar topic.

### CODE REVIEW

File: sensors/triggers/pulsar/pulsar.go [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+				if err != nil {
+					logger.Errorw("failed to get ClientCertPath from the volume", "error", err)
+					return nil, err
+				}
+				clientKeyPath, err = common.GetSecretVolumePath(pulsarTrigger.TLS.ClientKeySecret)
+				if err != nil {
+					logger.Errorw("failed to get ClientKeyPath from the volume", "error", err)
+					return nil, err
+				}
+			case pulsarTrigger.TLS.DeprecatedClientCertPath != "" && pulsarTrigger.TLS.DeprecatedClientKeyPath != "":
```
Comment (whynowy) [inline sensors/triggers/pulsar/pulsar.go:91]: Those deprecated fields were just removed.

**Review by whynowy — COMMENTED** [review]
It's great to see a Pulsar trigger implementation, thanks for your contribution!

Would you like to come up with dome docs under https://github.com/argoproj/argo-events/tree/master/docs/sensors/triggers, it could be either in this PR, or a follow up one.

File: sensors/triggers/pulsar/pulsar.go (hunk 1 above)
Comment (arnarg) [reply sensors/triggers/pulsar/pulsar.go]: I have removed that code now.

**Comment by arnarg** [comment]
@whynowy I will create some documentation tomorrow :)

**Comment by arnarg** [comment]
I have added some docs for the Pulsar trigger. It's a copy of the Kafka trigger adapted to my new Pulsar one.

**Review by whynowy — COMMENTED** [review]
LGTM, thanks!

**Review by whynowy — APPROVED** [review]
LGTM

**Pull request merged**